K8s-deployment/Charts/resource-server/templates/deployment.yaml

```
@@ -1,51 +1,51 @@
-{{- if .Values.async.enabled -}}
+{{- if .Values.resourceServer.enabled -}}
 apiVersion: {{ include "common.capabilities.deployment.apiVersion" . }}
 kind: Deployment
 metadata:
-  name: rs-async
+  name: rs-api-server
```

Change deployment name and labels to resource-server, to avoid confusion. Also matching whats there in HPA
```suggestion
  name: resource-server
```